src/pure.tsx

```diff
@@ -70,22 +76,30 @@ function renderHook(callback, { initialProps, wrapper } = {}) {
           {setValue}
         </TestHook>
       </Suspense>
-    )
+    ) as ReactElement 
+
+  
 
-  let testRenderer
-  act(() => {
+
+  // eslint-disable-next-line no-undef-init
+  let testRenderer: ReactTestRenderer | undefined
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    act(() => {
     testRenderer = create(toRender())
   })
-  const { unmount, update } = testRenderer
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { unmount, update } = testRenderer as ReactTestRenderer
 
   function rerenderHook(newProps = hookProps.current) {
     hookProps.current = newProps
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
     act(() => {
       update(toRender())
     })
   }
 
   function unmountHook() {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
     act(() => {
       removeCleanup(unmountHook)
       unmount()
```

Comment:
The Pure file needs to be heavily reviewed and ensure that the types are ideal for user interfacing.